group-coordinator/src/main/java/org/apache/kafka/coordinator/group/metrics/CoordinatorRuntimeMetrics.java

```diff
@@ -0,0 +1,244 @@
+
+    protected MetricsRegistry registry;
+    protected Metrics metrics;
+    public Sensor partitionLoadSensor;
+
+    /**
+     * Add metrics to the registries.
+     */
+    protected void initializeMetrics() {
+        registry.newGauge(numPartitionsInitial, new Gauge<Long>() {
+            @Override
+            public Long value() {
+                return NUM_PARTITIONS_INITIAL.get();
+            }
+        });
+
+        registry.newGauge(numPartitionsLoading, new Gauge<Long>() {
+            @Override
+            public Long value() {
+                return NUM_PARTITIONS_LOADING.get();
+            }
+        });
+
+        registry.newGauge(numPartitionsActive, new Gauge<Long>() {
+            @Override
+            public Long value() {
+                return NUM_PARTITIONS_ACTIVE.get();
+            }
+        });
+
+        registry.newGauge(numPartitionsClosed, new Gauge<Long>() {
+            @Override
+            public Long value() {
+                return NUM_PARTITIONS_CLOSED.get();
+            }
+        });
+
+        registry.newGauge(numPartitionsFailed, new Gauge<Long>() {
+            @Override
+            public Long value() {
+                return NUM_PARTITIONS_FAILED.get();
+            }
+        });
+
+        this.partitionLoadSensor.add(
+            metrics.metricName(
+                "partition-load-time-max",
+                metricsGroup(),
+                "The max time it took to load the partitions in the last 30sec"
+            ), new Max());
+        this.partitionLoadSensor.add(
+            metrics.metricName(
+                "partition-load-time-avg",
+                metricsGroup(),
+                "The average time it took to load the partitions in the last 30sec"
+            ), new Avg());
+    }
+
+    /**
+     * Retrieve the yammer metric name.
+     * @param name The name of the metric.
+     *
+     * @return The metric name.
+     */
+    protected abstract MetricName getMetricName(String name);
+
+    /**
+     * Retrieve the yammer metric name.
+     * @param group The metric group.
+     * @param type The metric type.
+     * @param name The name of the metric.
+     *
+     * @return The metric name.
+     */
+    protected static MetricName getMetricName(String group, String type, String name) {
+        return KafkaYammerMetrics.getMetricName(group, type, name);
+    }
+
+    /**
+     * @return The Kafka metrics group name.
+     */
+    protected abstract String metricsGroup();
+
+    @Override
+    public void close() {
+        Arrays.asList(
+            numPartitionsInitial,
+            numPartitionsLoading,
+            numPartitionsActive,
+            numPartitionsClosed,
+            numPartitionsFailed
+        ).forEach(registry::removeMetric);
+
+        metrics.removeSensor(partitionLoadSensor.name());
+    }
+
+    /**
+     * Increment the Initial partitions counter.
+     */
+    public void incrementNumPartitionsInitial() {
```

Comment:
Did you consider having something like `partitionStateChanged(old, new)`? The name is not so good but you get the idea.